api-documentation.md

```diff
@@ -2,41 +2,66 @@
 
 > Note: using [this markdown template](https://gist.github.com/iros/3426278).
 
-## **List links**
+## **GET links**
 
 Retrieve a list of links ordered by creation date.
 
-* **URL**:  `?do=api[TODO]`
+* **URL**: Endpoint discussion in #3.
 * **Method:** `GET`
-*  **URL Params**
+* **Path Parameters:** none
+* **URL Parameters**
 
    **Optional:**
  
     * `offset=[numeric]`
       Offset from which to start listing links (default: 0).
     * `limit=[numeric]`
-      Number of links to retrieve (default ?) or `all`.
-      
+      Number of links to retrieve (default 20) or `all`.
+    * `searchterm=[string]`
+      Search terms across all links fields.
+    * `searchtags=[string]`
+      Search for specifics tags. Tag list should be separated by a `+` delimitor.
+* **Body Parameters:** none
 
-* **Body Params:** none
-
-* **Success Response:**
-* **Content:** 
+* **Success Response:** `200`
+  **Content:** 
     ```json
     [
       { 
-        "todo": "link obj"
+        "id": "linkID",
+        "url": "Shaared URL",
+        "title": "link title",
+        "description": "link description",
+        "tags": [
+          "shaarli",
+          "php",
+          "api"
+        ],
+        "private": true,
+        "created": "2016-06-15T19:23:14+0200",
```

Comment:
Regarding timestamps, I've seen funny things over some APIs, where you need to multiply or divide the value of the returned Unix epoch to avoid time-travelling either to the Dark Ages™ or to the end of all known multiverses...

ISO8601 is indeed recommended for date/time representation:
- http://stackoverflow.com/questions/9581692/recommended-date-format-for-rest-get-api
- http://apiux.com/2013/03/20/5-laws-api-dates-and-times/

@nodiscc most decent programming languages have libraries for proper date/time parsing, that developers will use when they really care about timezones and Daylight Saving Time :)

![](https://imgs.xkcd.com/comics/iso_8601.png)